### PR TITLE
removes some uses of var in world

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

removes uses of for(var in world) for admin pm's, intercom debugging and player movement fixing
i was also gonna do this for (var/area in world) but i have to rethink what i did originally because i got confused

## Why It's Good For The Game

more performance or something

## Changelog
not player facing

### CODE REVIEW

File: code/modules/admin/verbs/adminpm.dm [hunk 1]
```diff
@@ -26,16 +26,16 @@
 			confidential = TRUE)
 		return
 	var/list/client/targets[0]
```
Comment (Fikou) [inline code/modules/admin/verbs/adminpm.dm:28]: im scared of this

File: code/modules/admin/verbs/adminpm.dm [hunk 2]
```diff
@@ -26,16 +26,16 @@
...
-	for(var/client/T)
-		if(T.mob)
-			if(isnewplayer(T.mob))
-				targets["(New Player) - [T]"] = T
-			else if(isobserver(T.mob))
-				targets["[T.mob.name](Ghost) - [T]"] = T
+	for(var/client/client in GLOB.clients)
```
Comment (Watermelon914) [inline code/modules/admin/verbs/adminpm.dm:29]:
```suggestion
	for(var/client/client as anything in GLOB.clients)
```

File: code/modules/admin/verbs/diagnostics.dm [hunk 3]
```diff
@@ -14,9 +14,7 @@
 	var/largest_click_time = 0
 	var/mob/largest_move_mob = null
 	var/mob/largest_click_mob = null
-	for(var/mob/M in world)
-		if(!M.client)
-			continue
+	for(var/mob/M in GLOB.player_list)
```
Comment (Watermelon914) [inline code/modules/admin/verbs/diagnostics.dm:17]:
```suggestion
	for(var/mob/M as anything in GLOB.player_list)
```
anything in GLOB.player_list is guaranteed to be a mob

File: code/modules/admin/verbs/diagnostics.dm [hunk 4]
```diff
@@ -14,9 +14,7 @@
 	var/largest_click_time = 0
 	var/mob/largest_move_mob = null
 	var/mob/largest_click_mob = null
-	for(var/mob/M in world)
-		if(!M.client)
-			continue
+	for(var/mob/M as anything in GLOB.player_list)
```
Comment (Kylerace) [inline code/modules/admin/verbs/diagnostics.dm:17]: fix name please

File: code/modules/admin/verbs/mapping.dm [hunk 5]
```diff
@@ -151,15 +138,16 @@ GLOBAL_LIST_EMPTY(dirty_vars)
...
-			for(var/turf/T in orange(7,I))
-				var/obj/effect/debugging/marker/F = new/obj/effect/debugging/marker(T)
-				if (!(F in view(7,I.loc)))
-					qdel(F)
+		for(var/frequency in GLOB.all_radios)
+			for(var/obj/item/radio/intercom/intercom in GLOB.all_radios[frequency])
+				for(var/turf/turf in orange(7,intercom))
+					var/atom/marker = new /obj/effect/abstract/marker/intercom(turf)
+					if(!(marker in view(7,intercom.loc)))
+						qdel(marker)
```
Comment (Kylerace) [inline code/modules/admin/verbs/mapping.dm:150]: why not just look for turfs in view(7, intercom.loc) in the first place?

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/misc.dm [hunk 6]
```diff
@@ -65,6 +65,9 @@
 /obj/effect/abstract/marker/at
 	name = "active turf marker"
 
+/obj/effect/abstract/marker/intercom
```
Comment (Kylerace) [inline code/game/objects/effects/misc.dm:68]: is there not a more fitting file to add to than a misc file

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/misc.dm (hunk 6 above)
Comment (Fikou) [reply code/game/objects/effects/misc.dm:68]: this is for consistency with the other abstract marker

File: code/modules/admin/verbs/adminpm.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/admin/verbs/adminpm.dm:28]: it can just be list()

**Pull request merged**